### ISSUE TITLE
SuggesterFilterBackend not returning expected SerializerMethod responses.

### ISSUE THREAD

**Issue by Ameer-Amr** [barseghyanartur/django-elasticsearch-dsl-drf#292]
SuggesterFilterBackend not returning SerializerMethod responses. it is perfectly ok with CompoundSearchFilterBackend.
but I am facing the issue with suggesters.
  
Here is my document.py

    `@registry.register_document
    class BrandDocument(Document):
        code = fields.TextField()
        name = fields.TextField(
            attr='name',
            fields={
                'suggest': fields.Completion(),
            }
        )
        name_ar = fields.TextField(
            attr='name_ar',
            fields={
                'suggest': fields.Completion(),
            }
        )
        brand_name_country = fields.TextField(
            attr='brand_name_country',
            fields={
                'suggest': fields.Completion(),
            }
        )
     .....
    `

Here is my serializers.py

    `class BrandDocumentSerializer(DocumentSerializer):
        ..........

        brand_name_country = serializers.SerializerMethodField()
    
        class Meta:
            document = BrandDocument
            fields = (
                'code',
                'name',
                'name_ar',
                'slug',
                'store',
                'label',
                'brand_name_country',
                'primary_category',
                'brand_denomination_range',
                'brand_image_gallery',
                'brand_image_data'
            )

          .................

        def get_brand_name_country(self, obj):
            return "{} {}".format(obj.name, obj.store['country']['name'])
    `

And here is my viewset

```
    class BrandDocumentView(DocumentViewSet):
        document = BrandDocument
        serializer_class = BrandDocumentSerializer
        filter_backends = [
            CompoundSearchFilterBackend,
            FilteringFilterBackend,
            SuggesterFilterBackend,
        ]
        pagination_class = BrandSearchPagination
        # renderer_classes = [JSONRenderer]
        # for specify search fields
        search_fields = {
            'name': {'fuzziness': 'AUTO'},
            'name_ar': {'fuzziness': 'AUTO'},
        }
    
        # Define filtering fields
        filter_fields = {
            'store': 'store.country.code'
        }
    
        # specify fields for suggesters with auto-completion
        suggester_fields = {
            'name_suggest': {
                'field': 'name.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
            },
            'name_ar_suggest': {
                'field': 'name_ar.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
            },
            'brand_name_country_suggest': {
                'field': 'brand_name_country.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
                'serializer_field': 'brand_name_country',
            },
        }

```

When I make a search it perfectly shows the result. like this:

![Screenshot 2022-07-15 at 12 31 40 PM](https://user-images.githubusercontent.com/102011541/179169745-e97d17bb-44c1-4e15-abda-6fc78c18faae.png)

but in suggestion i did't get the actual response:

![Screenshot 2022-07-15 at 12 35 53 PM](https://user-images.githubusercontent.com/102011541/179170394-189415d3-270d-4499-97a5-c263930a4b06.png)

look here, suggester response data are not the same as search response data. in search `'brand_name_country'` return its value. but in suggestion completion, it returns `Null`. and also in suggesters, it didn't return the images' absolute url's.

please have a look at this guys.

**Comment by Ameer-Amr** [comment]
> 

Oh.. could you please advise me if there is any hope for doing the same. or there is any way to pass params with SuggesterFilterBackend API..?

suppose I want to search for suggestions with 'amazon' in 'UAE' country, is that possible?
`...suggest/?name_suggest__completion=amazon&country=ae`

**Comment by barseghyanartur** [comment]
Sure. Look, suggester backend comes from a view mix-in. You override it and fix it as you want.

**Comment by Ameer-Amr** [comment]
> Sure. Look, suggester backend comes from a view mix-in. You override it and fix it as you want.

Thanks for the quick reply. could you please link an example of that or share that view mix-in quick link.?

**Comment by Ameer-Amr** [comment]
> Sure. Look, suggester backend comes from a view mix-in. You override it and fix it as you want.

Ok, thank you for your quick reply. I think I can manage my requirement with **category context suggesters**. and I tried to implement it in the following way. it responds 200 OK. but I didn't get the response data. is there anything that I am missing here?

in documents.py

```
        name = fields.TextField(
            attr='name',
            fields={
                'suggest': fields.CompletionField(),
                'suggest_context': fields.CompletionField(
                    contexts=[
                        {
                            "name": "country",
                            "type": "category",
                            "path": "store.country.code"
                        }
                    ]
                ),
            }
        )
        store = fields.ObjectField(
            properties={
                'country': fields.ObjectField(
                    properties={
                        'code': fields.TextField(),
                        'name': fields.TextField(
                            attr='name',
                            fields={
                                'suggest': fields.CompletionField()
                            }
                        )
                    },
                )
            },
        )
```

in my viewset:

    `   suggester_fields = {
            'name_suggest': {
                'field': 'name.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
            },
            'name_suggest_context': {
                'field': 'name.suggest_context',
                'default_suggester': SUGGESTER_COMPLETION,
                'completion_options': {
                    'category_filters': {
                        'name_suggest_country': 'country',
                    },
                },
                'options': {
                    'skip_duplicates': True,
                },
            },
            'name_ar_suggest': {
                'field': 'name_ar.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
            },
            'store_suggest': {
                'field': 'store.country.name.suggest',
                'suggesters': [
                    SUGGESTER_COMPLETION,
                ],
            },
        }`


and when i tried to request it with `.../suggest/?name_suggest_context=lulu&name_suggest_country=qa`, it returns this:

<img width="1174" alt="Screenshot 2022-07-16 at 2 57 41 PM" src="https://user-images.githubusercontent.com/90948477/179349125-f86eeb52-4005-408a-9302-93906700a333.png">

I reindexed my document but I got the same.

**Comment by barseghyanartur** [comment]
Yep, they are also supported. Check some examples [here](https://github.com/barseghyanartur/django-elasticsearch-dsl-drf/blob/master/src/django_elasticsearch_dsl_drf/filter_backends/suggester/native.py#L209).
